### PR TITLE
Migrate to AndroidX instead of Support library

### PR DESCRIPTION
**Pull request by voltazor**

### CODE REVIEW

**Comment by epool** [comment]
@voltazor could you please update `.travis.yml` according to make travis pass?

File: .travis.yml [hunk 1]
```diff
@@ -3,8 +3,8 @@ language: android
 android:
   components:
     - tools
-    - build-tools-27.0.3
-    - android-27
+    - build-tools-28.0.2
```
Comment (epool) [inline .travis.yml]: @voltazor according with travis, we don't need to change build tools version since we didn't touch it, just the android version.

**Comment by voltazor** [comment]
Now we have a problem with tests. I'll take a look, but I'm not familiar with Mockito, actually. So If you know a quick way to fix it, it would be great.

`org.mockito.exceptions.misusing.WrongTypeOfReturnValue: 
FragmentActivity$$EnhancerByMockitoWithCGLIB$$e1030257 cannot be returned by get()
get() should return RxPermissionsFragment
***
If you're unsure why you're getting above error read on.
Due to the nature of the syntax above problem might occur because:
1. This exception *might* occur in wrongly written multi-threaded tests.
   Please refer to Mockito FAQ on limitations of concurrency testing.
2. A spy is stubbed using when(spy.foo()).then() syntax. It is safer to stub spies - 
   - with doReturn|Throw() family of methods. More in javadocs for Mockito.spy() method.

	at com.tbruyelle.rxpermissions2.RxPermissionsTest.setup(RxPermissionsTest.java:65)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.robolectric.internal.SandboxTestRunner$2.evaluate(SandboxTestRunner.java:209)
	at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:109)
	at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:36)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.robolectric.internal.SandboxTestRunner$1.evaluate(SandboxTestRunner.java:63)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMainV2.main(AppMainV2.java:131)
`

**Comment by voltazor** [comment]
Seems like problem with robolectric shadows for new `androidx` `Fragment` and `FragmentActivity`

**Comment by voltazor** [comment]
I've used reflection for stubbing Activity in RxPermissionsFragment, because robolectric shadows can't work properly with androidx yet (https://github.com/robolectric/robolectric/issues/3985)

File: build.gradle [hunk 2]
```diff
@@ -19,14 +19,14 @@ allprojects {
...
-    compileSdkVersion = 27
+    compileSdkVersion = 28
     targetSdkVersion = compileSdkVersion
 
     rxJava = 'io.reactivex.rxjava2:rxjava:2.1.16'
-    supportLibraryVersion = '27.1.1'
-    appCompat = "com.android.support:appcompat-v7:$supportLibraryVersion"
-    supportAnnotations = "com.android.support:support-annotations:$supportLibraryVersion"
-    supportFragment = "com.android.support:support-fragment:$supportLibraryVersion"
+    androidxVersion = '1.0.0-alpha1'
```
Comment (XinyueZ) [inline build.gradle]: 1.0.0 is released

File: build.gradle [hunk 3]
```diff
@@ -4,7 +4,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.1.3'
+        classpath 'com.android.tools.build:gradle:3.1.4'
```
Comment (XinyueZ) [inline build.gradle]: 3.2.0 or 3.3.0-alpha12

**Comment by voltazor** [comment]
Hey @epool and @XinyueZ 
Any further change requests?

**Comment by voltazor** [comment]
Hey @epool 
Could you please merge pull request, if it's okay

**Comment by voltazor** [comment]
@epool any update?

**Comment by consp1racy** [comment]
I don't think it's the right time yet.

Jetifier provides a way to go from support to AndroidX. But there's no way to go in the opposite direction.

I'd save this PR for when the old support library numbering scheme is abandoned.

Thoughts?

**Comment by voltazor** [comment]
Google officially announced that support-library 28.0.0 - is the last feature release.
So, as for me, there is no reason to keep it. Migration guide from support-library to AndroidX is okay and migration itself required small amount of time

**Comment by vmeansdev** [comment]
Any ETA on this issue?

I was migrating my project to AndroidX and accidentally found out this library is not updated yet :(
I believe there lots of people who would like to use it with AndroidX together and it would be nice to have its support. Even wanted to make PR myself 😄 

Thank you

**Comment by voltazor** [comment]
@tbruyelle 
It's been a 3 month since AndroidX released, may be time to accept it has come?

**Comment by consp1racy** [comment]
It's been 3 months since AndroidX was released and Google thinks it's better for their upcoming library to be compatible with the old support library, so they're changing their dependecy *from AndroidX* to support library.

https://issuetracker.google.com/issues/120293333

What's the rush? Enable jetifier, which will convert all support library dependencies in your project to AndroidX, sit back, and relax.

https://developer.android.com/jetpack/androidx/migrate

And if you still have issues *in the IDE* after that, see if the project compiles anyway.

https://github.com/tbruyelle/RxPermissions/issues/263

**Comment by tbruyelle** [comment]
done with #325

**Pull request closed without merging**